### PR TITLE
WR-131 test(general): :white_check_mark: add tests for the lozenge component

### PR DESCRIPTION
**Pull request by ekansnek**
CURRENTLY ALL TESTS ARE PASSING

had to also add testID and accessibility states into the component so that I could test it (doesn't affect function), can comment out if not testing

---

<details open><summary><strong>Checklist</strong></summary>

- [x] My PR title is prefixed with WR-XX
- [ ] If I made a frontend change, I have included videos/screenshots of the changes on both iOS and Android
</details>

### CODE REVIEW

File: app/components/Tests/Lozenge.test.tsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+    const { getByText: getByText9 } = renderWithProviders(<Lozenge type="declined" />)
+    expect(getByText9("Declined")).toBeTruthy()
+
+    const { getByText: getByText10 } = renderWithProviders(<Lozenge type="assigned" />)
+    expect(getByText10("Assigned")).toBeTruthy()
+
+    const { getByText: getByText11 } = renderWithProviders(<Lozenge type="openShift" />)
+    expect(getByText11("OpenShift")).toBeTruthy()
+  })
```
Comment (rcchao) [inline app/components/Tests/Lozenge.test.tsx:55]: Let's not repeat ourselves.

Separate out and export LOZENGE_CONFIG in the `Lozenge.tsx` file like this:
```
export const LOZENGE_CONFIG: Record<LozengeType, LozengeConfig> = {
```

Then use it here like this:


```suggestion
it("renders correct text for each type", () => {
  Object.keys(LOZENGE_CONFIG).forEach((type) => {
    const { getByText } = renderWithProviders(<Lozenge type={type as any} active />)
    const expectedText = LOZENGE_CONFIG[type as keyof typeof LOZENGE_CONFIG].text
    expect(getByText(expectedText)).toBeTruthy()
  })
})
```

File: app/components/Lozenge.tsx [hunk 2]
```diff
@@ -122,13 +122,18 @@ export const Lozenge = ({ type, active = false, onPress }: LozengeProps) => {
       alignSelf="flex-start"
       disabled={!active}
       backgroundColor={buttonBgColor}
+      accessibilityRole="button"
```
Comment (rcchao) [inline app/components/Lozenge.tsx:125]: Is this line necessary?

`      accessibilityRole="button"`

**Review by rcchao — CHANGES_REQUESTED** [review]
Minor changes, good job working out tests! 

I must say though this is definitely overkill for a Lozenge component. For future components try to keep the test suite more concentrated, maybe just test the happy case (everything works as expected) and if it's a more complicated or crucial component, a couple plausible edge cases. This way we can make sure our efforts are proportional to what we need to achieve. Great work still!
